corehq/ex-submodules/pillowtop/processors/form.py

```diff
@@ -104,4 +118,14 @@ def mark_latest_submission(domain, user_id, app_id, build_id, version, metadata,
         last_submission.build_version = app_version_info.build_version
         last_submission.commcare_version = app_version_info.commcare_version
 
+        if app_version_info.build_version:
+            update_latest_builds(user, app_id, received_on_datetime, app_version_info.build_version)
+
+        if _last_submission_needs_update(user.reporting_metadata.last_submission_for_user,
+                                         received_on_datetime,
+                                         app_version_info.build_version,
+                                         app_version_info.commcare_version):
```

Comment:
I think I am missing something. The first if can be true if the second one isnt (there could be a submission for a different app that is the global most recent even if this is the app most recent), but not the reverse (if this is the global most recent it is also necessarily the app most recent). Because of that we should need to check both cases and with the broader if first. Even if we are saving the user we are not necessarily updating the global last submission